scRNA-seq/pre-processing_scRNA-seq.md

```diff
@@ -0,0 +1,149 @@
+
+## Steps for Processing scRNA-seq Data:
+
+### Step 0: Obtaining the data
+10X Genomics has plenty of [datasets](https://support.10xgenomics.com/single-cell-gene-expression/datasets)
+available to play around with.
+For this example, we will use the [1k pbmc](https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/pbmc_1k_v2)
+(peripheral blood mononuclear cells).
+We will process a fastq file of this as an example.  
+For practical purposes as far as time constraints, we have subset this fastq file
+and have it ready for you to process but with only the first 3 million reads.
+
+*Note*: depending on the state of the data you are working with, ie. if you have
+a `.bcl` file, you will need to use `CellRanger` with their `mkfastq` command to
+make fastq files from this.
+However, because most publicly available data is in fastq format and any
+core you might be working with will also likely provide you with fastq files,
+we are starting from this point.  
+
+### Step 1: Set up your output directory
+As described in a previous module, we'll make a directory from command line
+for us to store our output data in.
+```
+mkdir alevin_output  
+```
+
+### Step 2: Index the human transcriptome with Salmon
+Before you can quantify with Salmon and
+[Alevin](https://www.biorxiv.org/content/10.1101/335000v2), we need a transcriptome
+to be indexed.
+You can use the same trancriptome index as was used for bulk-rna-seq, however,
+due to the smaller pieces and amounts of single cell RNA-seq as opposed to bulk,
+you may want to build the index with a smaller `-k`.
+In this instance, we used a `-k` of 23 using the ensemble transcriptome.
+
+In the interest of time, we have already run the command below and have the index
+built for you.
+But for your own reference, here is how you'd do it yourself:
+```
+salmon --threads=16 --no-version-check index \
+  -t Homo_sapiens.GRCh38.cdna.all.fa.gz \
+  -i human_index \
+  -k 23
+```
+
+### Step 3: For each sample, run [Alevin](https://www.biorxiv.org/content/10.1101/335000v2)
+for quantification
+From the command line, running Alevin is not too much different from running
+Salmon for bulk rna-seq. You'll recognize a lot of these options as the same.
+In this instance, files with `R1` contain the barcodes for the cells as well as
+the Unique Molecular Identifiers while `R2` files contain the full reads for that sample.  
+
+#### `-l`
+The ISR library type is what is recommended for single cell data quant.
+
+#### `--chromium`
+Because we are using 10X chromium data, we have to use this flag. However,
+Drop-seq data is also supported, and in this case you would use a `--dropseq`
+flag instead of this.
+
+#### `--tgMap`
+This is needed to supply a trasncript to gene key that Alevin will use to
+quantify the genes. For our example, we've premade the file `genes_2_tx.tsv` from
+the ensembl transcriptome that we indexed above. The file has to be a .tsv file.
```

Comment:
```suggestion
the Ensembl transcriptome that we generated with `salmon index` above. The file has to be a .tsv file.
```

You say csv (no period below), I think the preferred style for these would be TSV and CSV.